Cabal/Distribution/Simple/LocalBuildInfo.hs

```diff
@@ -266,24 +268,37 @@ depLibraryPaths inplace relative lbi clbi = do
     -}
         getLibDir sub_clbi
           | inplace    = componentBuildDir lbi sub_clbi
-          | otherwise  = libdir (absoluteComponentInstallDirs pkgDescr lbi (componentUnitId sub_clbi) NoCopyDest)
+          | otherwise  = dynlibdir (absoluteComponentInstallDirs pkgDescr lbi (componentUnitId sub_clbi) NoCopyDest)
```

Comment:
Well, that entire function is there to find the locations of dynamic libraries. Since `libdir` now only contains the location of `static` libraries, we should use `dynlibdir`.